src/frontend/components/InstructorDashboardVOD/widgets/BaseUploadTimedTextTracks/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { Box, Button } from 'grommet';
+import React, { useEffect, useRef, useState } from 'react';
+import { defineMessages, useIntl } from 'react-intl';
+
+import { DashboardVideoLiveItemList } from 'components/DashboardVideoLiveControlPane/customs/DashboardVideoLiveItemList';
+import { useUploadManager } from 'components/UploadManager';
+import { createTimedTextTrack } from 'data/sideEffects/createTimedTextTrack';
+import { useTimedTextTrack } from 'data/stores/useTimedTextTrack';
+import { useTimedTextTrackLanguageChoices } from 'data/stores/useTimedTextTrackLanguageChoices';
+import { LanguageChoice } from 'types/LanguageChoice';
+import { modelName } from 'types/models';
+import { timedTextMode, uploadState } from 'types/tracks';
+import { Nullable } from 'utils/types';
+import { LanguageSelect } from './LanguageSelect';
+import { TimedTextTrackItem } from './TimedTextTrackItem';
+
+const messages = defineMessages({
+  noUploadFilesMessage: {
+    defaultMessage: 'No uploaded files',
+    description: 'Message displayed when there is no uploaded document.',
+    id: 'components.BaseUploadTimedTextTracks.noUploadFilesMessage',
+  },
+  uploadButtonLabel: {
+    defaultMessage: 'Upload file',
+    description: 'Label of the button used to upload the timed text file.',
+    id: 'components.BaseUploadTimedTextTracks.uploadButtonLabel',
+  },
+});
+
+interface BaseUploadTimedTextTracksProps {
+  timedTextModeWidget: timedTextMode;
```

Comment:
```suggestion
  timedTextMode: timedTextMode;
```

I don't see the interest to add this suffix. What does it bring ?

Reply:
It does not bring anything, but it prevents `timedTextMode` import to be shadowed by the `timedTextMode` variable

Reply:
I hate this rule...